linux/LinuxProcessList.c

```diff
@@ -1824,8 +1824,33 @@ static inline double LinuxProcessList_scanCPUTime(ProcessList* super) {
    return period;
 }
 
+static bool cpuIsOnline(unsigned int cpuId) {
+      /* if there's no sysfs file for the CPU node indicating online/offline status,
+       * the CPU is probably online, so assume online by default
+       */
+      bool online = 1;
+      char pathBuffer[64];
+
+      xSnprintf(pathBuffer, sizeof(pathBuffer), "/sys/devices/system/cpu/cpu%u/online", cpuId);
+
+      FILE* file = fopen(pathBuffer, "r");
+      unsigned int readval;
+      if (file) {
+         if (fscanf(file, "%u", &readval) == 1) {
+            online = readval;
+         }
+         fclose(file);
+      }
+
+      return online;
+}
+
+static int getConfiguredCPUCount(void) {
+   return (int) sysconf(_SC_NPROCESSORS_CONF);
+}
+
 static int scanCPUFreqencyFromSysCPUFreq(LinuxProcessList* this) {
-   unsigned int cpus = this->super.cpuCount;
+   int configuredCPUs = getConfiguredCPUCount();
```

Comment:
Try to cache that value and avoid requesting it on every run; occasional refresh is no problem though.